src/shared/components/palette/complex/ccd-cyapage-label-filter.pipe.ts

```diff
@@ -0,0 +1,28 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { CaseField } from '../../../domain';
+
+@Pipe({
+  name: 'ccdCYAPageLabelFilter'
+})
+export class CcdCYAPageLabelFilterPipe implements PipeTransform {
+
+  public transform(caseFields: CaseField[]): CaseField[] {
+    return caseFields.map((caseField: CaseField) => {
+      if (caseField.field_type.collection_field_type &&
+        caseField.field_type.collection_field_type.complex_fields &&
+        caseField.field_type.collection_field_type.complex_fields.length) {
+        caseField.field_type.collection_field_type.complex_fields = this.transform(
+          caseField.field_type.collection_field_type.complex_fields
+        )
+      }
+      if (caseField.field_type.complex_fields && caseField.field_type.complex_fields.length) {
+        caseField.field_type.complex_fields = this.getNonLabelComplexFieldType(caseField.field_type.complex_fields);
+      }
+
+      return caseField;
+    });
+  }
+
+  private getNonLabelComplexFieldType = (complexFields: CaseField[]): CaseField[] =>
```

Comment:
_Question:_
Any particular reason why this is an arrow function rather than just a function?

Reply:
Arrow functions are best for callbacks which is why i believe dan introduce it.